core/ipc/rpc/kafka/src/main/java/org/opennms/core/ipc/rpc/kafka/KafkaRpcServerManager.java

```diff
@@ -258,40 +256,24 @@ public void run() {
                             if (rpcMessage.getTotalChunks() > 1) {
                                 // Handle multiple chunks
                                 boolean allChunksReceived = handleChunks(rpcMessage);
-                                if(!allChunksReceived) {
+                                if (!allChunksReceived) {
                                     continue;
                                 }
                                 rpcContent = messageCache.get(rpcId);
                                 //Remove rpcId from cache.
                                 messageCache.remove(rpcId);
                                 currentChunkCache.remove(rpcId);
                             }
-                            //Build child span from rpcMessage and start minion span.
-                            Tracer.SpanBuilder spanBuilder = buildSpanFromRpcMessage(rpcMessage);
-                            Span minionSpan = spanBuilder.start();
-
-                            RpcRequest request = module.unmarshalRequest(rpcContent.toStringUtf8());
-                            setTagsOnMinion(rpcMessage, request, minionSpan);
-
-                            CompletableFuture<RpcResponse> future = module.execute(request);
-                            future.whenComplete((res, ex) -> {
-                                final RpcResponse response;
-                                if (ex != null) {
-                                    // An exception occurred, store the exception in a new response
-                                    LOG.warn("An error occured while executing a call in {}.", module.getId(), ex);
-                                    response = module.createResponseWithException(ex);
-                                    minionSpan.log(ex.getMessage());
-                                    minionSpan.setTag(TAG_RPC_FAILED, "true");
-                                } else {
-                                    // No exception occurred, use the given response
-                                    response = res;
-                                }
-                                // Finish minion Span
-                                minionSpan.finish();
-                                sendResponse(rpcId, response);
-                            });
+                            final RpcModule module = modulesById.get(rpcMessage.getModuleId());
+                            if(module == null) {
```

Comment:
This is a problem - if a Minion doesn't have a module loaded it still consumes the relevant requests and drops them silently.

Reply:
yeah.  If you are considering a scenario where certain modules are loaded on one minion and rest of the modules are loaded on another minion, that may not work with this approach.

Reply:
I realized that we can't disable `RPC` as that would disable Echo module as well which is needed  for maintaining minion status.  We can still create a topic specific to Echo and subscribe to it by default.  subscribe to generic `rpc-request` topic only if at least one of the other `rpc` modules are loaded.  Let me know what do you think.

Reply:
Or introduce two more Minion Status, `RPC_UP`, `SINK_UP`